plutus-contract/src/Wallet/Emulator/MultiAgent.hs

```diff
@@ -302,12 +303,17 @@ emulatorStateInitialDist mp = emulatorStatePool [EmulatorTx tx] where
             , txData = mempty
             }
     -- See [Creating wallets with multiple outputs]
-    mkOutputs (key, vl) = mkOutput key <$> splitHeadinto10 (Wallet.splitOffAdaOnlyValue vl)
+    mkOutputs (key, vl) = mkOutput key <$> splitHeadinto10 (splitOffAdaOnlyValue vl)
     splitHeadinto10 []       = []
     splitHeadinto10 (vl:vls) = replicate (fromIntegral count) (Ada.toValue . (`div` count) . Ada.fromValue $ vl) ++ vls
         where
             -- Make sure we don't make the outputs too small
             count = min 10 $ Ada.fromValue vl `div` minAdaTxOut
+    -- | Split value into an ada-only and an non-ada-only value, making sure each has at least minAdaTxOut.
+    splitOffAdaOnlyValue :: Value -> [Value]
```

Comment:
Why not `... -> (Value, Value)` here? Seems like we always expect a list of 2 values.